### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by Bupilipili**
### In this branch

- Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make my HTTP requests to the backend
- Fetched my books from the Bookstore API after the component has rendered
- Refactored my add book and remove book features to persist my changes in the server
- My application behaves the same way after the refactoring.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -18,22 +20,27 @@ const initialState = [
...
     category: 'Nonfiction',
-  },
-];
+  },];
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/abc123/books');
+  return response.data;
+});
+
```
Comment (FrederickMih) [inline src/redux/books/booksSlice.js]: - [x] Hi @Bupilipili, now we should connect this external API to our existing React-Redux application by refactoring codes in such a way that adding and removing books is done on a remote server(API). You did a nice job creating  the fetchBooks action creator but this should exist in the bookSlice. fetchBooks action creator should actually updates the state of bookSlice and make sure useEffect is use when dispatching it in the BookList component. addBook and removeBook should be refactor to return a promise, it should add, remove new book through the API and authomatically dispatch an action type that will add, remove a book to the books list and updates the state. Please kindly fix :wrench:  the issue.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @Bupilipili, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Nice job for implementing the fetchBooks action creator :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (Bupilipili) [reply src/redux/books/booksSlice.js]: Thanks for the insight 💯 am on it 😃

**Review by Davidosky007 — APPROVED** [review]
Hi @Bupilipili 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
*  React and Redux connected properly 👍 
* Store, actions and reducers in React used properly 👍 
* API works properly 👍  
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bupilipili** [comment]
Thanks 👍 💯 .

**Pull request merged**